examples/mnist_01.py

```diff
@@ -51,23 +52,35 @@ class MNISTNetConf:
 
 
 class Net(nn.Module):
-    def __init__(self):
+    def __init__(self, input_shape, output_shape, cfg):
         super(Net, self).__init__()
         self.conv1 = nn.Conv2d(1, cfg.model.conv1_out_channels, 3, 1)
         self.conv2 = nn.Conv2d(
             cfg.model.conv1_out_channels, cfg.model.conv2_out_channels, 3, 1
         )
         self.dropout1 = nn.Dropout2d(cfg.model.dropout1_prob)
         self.dropout2 = nn.Dropout2d(cfg.model.dropout2_prob)
-        self.fc1 = nn.Linear(9216, cfg.model.fc_hidden_features)
-        self.fc2 = nn.Linear(cfg.model.fc_hidden_features, 10)
+
+        conv_out_shape = self._compute_conv_out_shape(input_shape)
+        linear_in_shape = torch.prod(conv_out_shape)
+
+        self.fc1 = nn.Linear(linear_in_shape, cfg.model.fc_hidden_features)
+        self.fc2 = nn.Linear(cfg.model.fc_hidden_features, torch.prod(output_shape))
+
+    def _compute_conv_out_shape(self, input_shape):
+        dummy_input = torch.zeros(input_shape)
+        with torch.no_grad():
+            x = self.conv1(dummy_input)
+            x = self.conv2(x)
+            dummy_output = F.max_pool2d(x, cfg.model.maxpool1_kernel)
+        return dummy_output.shape()
 
     def forward(self, x):
         x = self.conv1(x)
         x = F.relu(x)
         x = self.conv2(x)
         x = F.relu(x)
-        x = F.max_pool2d(x, 2)
+        x = F.max_pool2d(x, cfg.model.maxpool1_kernel)
```

Comment:
Did you run this? where is cfg coming from?

Reply:
I've changed it since. Was just quickly prototyping some possible ideas and accidentally left the cfg for maxpool in there. Decided it's better to move it out of forward and use the standard class rather than the functional version.

Reply:
In Hydra, I have tests that executes the example applications. You can do that here too.

Reply:
Issue #31